### PR TITLE
Implemented response headers when using `XMLHttpRequest`

### PR DESCRIPTION
**Pull request by mikedriver**
I think perhaps these were left out by mistake?

### CODE REVIEW

**Comment by vjeux** [comment]
Thanks! We never needed them so far so haven't implemented it.

I'll pull this out on monday (or before if I'm motivated)

**Comment by mikedriver** [comment]
Didn't notice that `getAllResponseHeaders()` should return a string.  According to MDN it should also return `null` if no headers rather than a blank string.  However I left it as a blank string instead because I noticed that the fetch library is doing `xhr.getAllResponseHeaders().trim().split('\n')`  which would otherwise throw.

File: Libraries/Network/RCTDataManager.m [hunk 1]
```diff
@@ -57,9 +57,11 @@ - (void)executeQuery:(NSString *)queryType
         } else {
           encoding = NSUTF8StringEncoding;
         }
-        int responseCode = (int)[((NSHTTPURLResponse *)response) statusCode];
+        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse*)response;
+        NSDictionary *headerDict = [httpResponse allHeaderFields];
+        int responseCode = (int)[httpResponse statusCode];
```
Comment (nicklockwood) [inline Libraries/Network/RCTDataManager.m]: This should be NSInteger, which is what the method actually returns . Not sure why we'd cast to an int (I appreciate that this was pre-existing, and not something you changed).

**Comment by crockeo** [comment]
@vjeux any update on merging this pull request?

**Comment by nicklockwood** [comment]
Just looking into it now. Looks like it failed to land due to a unit test failure on internal CI. Probably a flaky test. I'll try re-submitting it.

**Pull request closed without merging**